exp/controllers/azuremachinepool_controller.go

```diff
@@ -210,25 +211,57 @@ func (ampr *AzureMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	logger = logger.WithValues("cluster", cluster.Name)
 
-	logger = logger.WithValues("AzureCluster", cluster.Spec.InfrastructureRef.Name)
-	azureClusterName := client.ObjectKey{
-		Namespace: azMachinePool.Namespace,
-		Name:      cluster.Spec.InfrastructureRef.Name,
-	}
-	azureCluster := &infrav1.AzureCluster{}
-	if err := ampr.Client.Get(ctx, azureClusterName, azureCluster); err != nil {
-		logger.V(2).Info("AzureCluster is not available yet")
-		return reconcile.Result{}, nil
-	}
+	var clusterScope azure.ClusterScoper
+	var groupScope groups.GroupScope
```

Comment:
Offering this in case it makes anything easier if we know that these will always refer to the same thing:
```suggestion
	var clusterScope interface {
		azure.ClusterScoper
		groups.GroupScope
	}
```
Or perhaps a new typedef local to this package since it looks like that combo may be useful elsewhere.

Reply:
This one was a pain... I tried to do this before but I ran into a tangle of import cycles between azure, mock_azure, groups, and aso packages 

Because of the refactor with the helper function it's now a lot cleaner to have a separate interface, I put it in the controllers helpers since it's used there and that way it avoids all the imprt cycles